### PR TITLE
Fail fast an ingester if unable to load existing TSDBs

### PR DESCRIPTION
**Pull request by pracucci**
**What this PR does**:
Tonight we had an issue in one ingester which had TSDB head chunks corrupted (root cause will be discussed separately). When a similar issue happen, the ingester skips the corrupted TSDB at startup, it joins the ring with `ACTIVE` state and, as soon as receive any write request from the tenant with the corrupted TSDB it will try to reopen the TSDB for *every single write request*. This leads to an undesirable situation, which will soon get the ingester to get killed (due to OOM).

In this PR I'm proposing to fail fast the ingester if unable to load an existing TSDB. It's a loud and clear signal to the Cortex cluster operator and, in my opinion, it's better to fail fast an ingester before it start receiving write requests instead of having it failing few minutes after running.

**Which issue(s) this PR fixes**:
N/A

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/ingester/ingester_v2.go [hunk 1]
```diff
@@ -1057,8 +1064,13 @@ func (i *Ingester) openExistingTSDB(ctx context.Context) error {
...
 
-	err := filepath.Walk(i.cfg.BlocksStorageConfig.TSDB.Dir, func(path string, info os.FileInfo, err error) error {
+	// Keep track of all errors that could occur.
+	errs := tsdb_errors.MultiError{}
+	errsMx := sync.Mutex{}
+
+	walkErr := filepath.Walk(i.cfg.BlocksStorageConfig.TSDB.Dir, func(path string, info os.FileInfo, err error) error {
 		if err != nil {
+			level.Warn(util.Logger).Log("msg", "skipped filesystem entry when looking for existing TSDB to open", "path", path, "err", err)
 			return filepath.SkipDir
```
Comment (pstibrany) [inline pkg/ingester/ingester_v2.go:1074]: If there is any error traversing the tree, shouldn't we return such error?  (Esp. if we're halting ingester if we fail to open TSDB)

File: pkg/ingester/ingester_v2.go [hunk 2]
```diff
@@ -1114,14 +1131,23 @@ func (i *Ingester) openExistingTSDB(ctx context.Context) error {
 		return filepath.SkipDir // Don't descend into directories
 	})
 
+	if walkErr != nil {
```
Comment (pstibrany) [inline pkg/ingester/ingester_v2.go:1134]: This cannot happen currently, since walkFn will filter out any error. (see other comment)

**Review by pstibrany — APPROVED** [review]
LGTM, but please take a look at comment.

File: pkg/ingester/ingester_v2.go (hunk 1 above)
Comment (pracucci) [reply pkg/ingester/ingester_v2.go:1074]: Yes, better to do error out. Done.

File: pkg/ingester/ingester_v2.go (hunk 2 above)
Comment (pracucci) [reply pkg/ingester/ingester_v2.go:1134]: Given I addressed the other comment, why can't happen? The `Walk()` interrupts and returns error as soon as we return error. Errors returned by `Walk()` itself are not filtered again via `Walk()`.

Comment (pstibrany) [reply pkg/ingester/ingester_v2.go:1134]: Now it can.

**Review by pstibrany — APPROVED** [review]

File: pkg/ingester/ingester_v2.go (hunk 2 above)
Comment (pracucci) [reply pkg/ingester/ingester_v2.go:1134]: Even before it could, in case `os.Open(path)` or `f.Readdirnames(1)` failed.

Comment (pstibrany) [reply pkg/ingester/ingester_v2.go:1134]: You are right.

**Pull request merged**